### PR TITLE
use content scripts instead of preload

### PR DESCRIPTION
**Pull request by bridiver**
https://github.com/brave/electron/pull/10

### CODE REVIEW

**Comment by bbondy** [comment]
This currently breaks preferences

**Comment by bridiver** [comment]
Probably because I removed the message event listener. It wasn't needed for the webview preload anymore and I didn't think about the preferences

File: js/components/frame.js [hunk 1]
```diff
@@ -54,12 +54,6 @@ class Frame extends ImmutableComponent {
 
   componentDidMount () {
     this.updateWebview()
-    // forward postMessage events from webview to webContents
-    window.addEventListener('message', function (event) {
```
Comment (bridiver) [inline js/components/frame.js]: This probably broke the preferences. If we are going to keep using contentWindow for them we should lock down the origin for it

File: js/components/frame.js [hunk 2]
```diff
@@ -54,12 +54,6 @@ class Frame extends ImmutableComponent {
...
-      if (this.webview.getAttribute('src').match(event.origin)) {
```
Comment (bbondy) [inline js/components/frame.js]: Yep per slack match origin will also match things like brianbondy.com#file:// and allow it. So this should be startsWith I think and also I'm not sure how to lock this to a full path and not just the file protocol.

Comment (bridiver) [inline js/components/frame.js]: does setting document.domain work?

Comment (bbondy) [inline js/components/frame.js]: document.domain will be set to blank and also I think origin for a file URL will always return file://

**Comment by bbondy** [comment]
Merged this and will fix preferences in master

**Pull request merged**